samples/OrdersApi/Consumer.Tests/OrdersClientTests.cs

```diff
@@ -84,6 +84,58 @@ await this.pact.VerifyAsync(async ctx =>
             });
         }
 
+        [Fact]
+        public async Task GetOrdersAsync_WhenCalled_ReturnsMultipleOrders()
+        {
+            var expected1 = new OrderDto(1, OrderStatus.Pending, new DateTimeOffset(2023, 6, 28, 12, 13, 14, TimeSpan.FromHours(1)));
+            var expected2 = new OrderDto(2, OrderStatus.Pending, new DateTimeOffset(2023, 6, 29, 12, 13, 14, TimeSpan.FromHours(1)));
+
+            this.pact
+                .UponReceiving("a request for multiple orders by id")
+                .Given("orders with ids {ids} exist", new Dictionary<string, string> { ["ids"] = "1,2" })
+                .WithRequest(HttpMethod.Get, "/api/orders/many/1,2")
+                .WithHeader("Accept", "application/json")
+                .WillRespond()
+                .WithStatus(HttpStatusCode.OK)
+                .WithJsonBody(Match.ArrayContains(new dynamic[]
+                {
+                    new
```

Comment:
issue: I think for this example to 'come to life' then we'd really need it to match multiple different possible variants

Perhaps if there were different order types or something so that they could have different fields on them, then the matching rules could reflect those variants. Like a fulfilled order has an extra date on it for when it was completed, or something like that.